src/t_hash.c

```diff
@@ -1634,12 +1647,14 @@ void genericHgetallCommand(client *c, int flags) {
     }
 
     hashTypeResetIterator(&hi);
-    /* Make sure we returned the right number of elements. */
-    if (flags & OBJ_HASH_FIELD && flags & OBJ_HASH_VALUE) {
-        setDeferredMapLen(c, replylen, count /= 2);
-        count /= 2;
-    } else {
-        setDeferredArrayLen(c, replylen, count);
+    /* In case of deferred reply, make sure we returned the right number of elements. */
+    if (replylen) {
+        if (flags & OBJ_HASH_FIELD && flags & OBJ_HASH_VALUE) {
+            setDeferredMapLen(c, replylen, count /= 2);
+            count /= 2;
```

Comment:
This looks like an old typo that was preserved in this change.

Dividing count by two twice doens't make sense. `count` is a local variable and this is the last use, so I believe a simple `count / 2` is correct and more clear.

```suggestion
            setDeferredMapLen(c, replylen, count / 2);
```